plugins/rest-api.php

```diff
@@ -0,0 +1,375 @@
+<?php
+
+/*
+ * Add support for caching REST API v2+ endpoints
+ *
+ * URLs without parameters -- e.g., https://example.org/wp-json/wp/v2/posts/ -- can be served cached files by
+ * mod_rewrite. URLs with parameters -- e.g., https://example.org/wp-json/wp/v2/posts?per_page=10 -- are served
+ * cached files by PHP.
+ *
+ */
+
+// Prevent direct access
+if ( ! defined( 'WPCACHEHOME' ) ) {
+	return;
+}
+
+/*
+ * Future development:
+ *
+ * See if there's a way to serve gzipped json while still sending the correct content-type header
+ *      can use Apache's ModMime?
+ *          Add filter to $gziprules in wpsc_get_htaccess_info
+ *          Add `<FilesMatch "\.json(\.gz)?$">` to cache/.htaccess
+ *          Add `RewriteCond %{HTTP:Accept-Encoding} gzip` rules to /.htaccess
+ *		maybe detect if server is configured to send application/json for gzipped
+ *          if it's not, then don't generate the corresponding rules?
+ *      could maybe put AddType into htaccess to set gzip mime type
+ *          thought not all servers supported, but w3 total cache does it. maybe they detect support first?
+ *
+ * When disabling the plugin, clear all cached files to get rid of the API cache, then run the preloader to regenerate everything else (if it's enabled)
+ *
+ * Need to update the cache when underlying data changes, rather than just waiting for manual expiration?
+ *      would be bad if deleted post continue to show up in cache, etc
+ *      how to tell that updating a post or category maps to an API endpoint?
+ *      worst case, could just flush all endpoint cache files when any post/comment/taxterm/etc is updated
+ *
+ * JSON spec doesn't allow comments, but still want the helpful info from $wp_super_cache_comments
+ *    so instead, maybe send X-WPSC-REST headers with that info
+ *    maybe can't get that b/c has to be set in phase2, but can at least output whether loading from cache or not. already has a header for that, though
+ *    could store in meta files, like headers
+ *
+ * Add support for preloading
+ *
+ * Add .json tests to the Cache Tester button on Settings > Easy ?
+ *      maybe send a test json response to make sure the header is getting set correctly
+ *      if not, tell user to ask host to configure apache to send application/json for .json files
+ *
+ * Run `wpsc_rest_save_url_prefix` once during WPSC setup, and then on a cron after that
+ *      that's cleaner/faster than loading on every request
+ */
+
+/**
+ * Load the plugin
+ */
+function wpsc_rest_bootstrap() {
+	global $cache_rest_api;
+
+	// Always register the plugin settings
+	add_cacheaction( 'cache_admin_page', 'wpsc_rest_render_plugin_settings' );
+
+	/*
+	 * Disable the rest of the plugin if that was explicitly configured.
+	 * Otherwise, leave it enabled by default and set a canonical value for `$cache_rest_api`.
```

Comment:
I'm fine with making this default. Decisions not options and if someone is using the API, it makes sense to me to cache it from the onset.